pkg/eventsapi/v2.go

```diff
@@ -17,18 +17,29 @@ type EventV2 struct {
 	Links       []LinkV2  `json:"links,omitempty"`
 }
 
-func (e EventV2) GetRoutingKey() string {
+func (e *EventV2) GetRoutingKey() string {
```

Comment:
My thought process was this: `AddCustomDetail` needs to be passed a pointer reference because it's mutating one of the fields in the underlying struct (which I think is correct). My two options were to either make `AddCustomDetail` return a new event and keep the data immutable, which I didn't like as much as passing the pointer reference, and then from there I changed the rest of of the methods to take a pointer reference for consistency.